projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/Login2.tsx

```diff
@@ -0,0 +1,148 @@
+    }
+
+    closeAuthWindow = () => {
+        if (this.authWindow && !this.authWindow.closed) {
+            this.authWindow.close();
+        }
+    }
+
+    handleURLInput = async (e: { target: { value: any; }; }) => {
+        const redirectURI = e.target.value;
+        const code = new URL(redirectURI).searchParams.get('code');
+        if (code) {
+            try {
+                await requestAPI<any>(
+                    'login?code=' + code,
+                    {
+                        method: 'GET'
+                    }
+                );
+                this.closeAuthWindow();
+                this.setState({ authStatus: AuthStatus.SUCCESS });
+            } catch (error) {
+                this.closeAuthWindow();
+                console.error("Error during authentication:", error);
+                this.setState({ authStatus: AuthStatus.FAILURE });
+            }
+        }
+    }
+
+    render() {
+        return (
+            <div className={LOGIN2_DIALOG_CLASS}>
+                {this.state.authStatus === AuthStatus.NONE &&
+                    <div>
+                        <p>Authenticate process will start. After you clik a new window will be opened.</p>
+                        <ul>
+                            <li>Please <b>wait</b> for the authentication window to redirect you back to 127.0.0.1.</li>
+                            <li><b>Ignore</b> the failed connection error message.</li>
+                            <li>And then <b>copy and paste</b> the full URL you are redirected to.</li>
+                        </ul>
+
+                        <button className={LOGIN2_BUTTON_CLASS} onClick={this.openAuthWindow}>Click here to start.</button>
+                    </div>
+                }
+                {this.state.authStatus === AuthStatus.WINDOW_OPENED &&
+                    <div>
+                        <p>A new authentication window has been opened.</p>
+                        <ul>
+                            <li>Please <b>wait</b> for the authentication window to redirect you back to 127.0.0.1.</li>
+                            <li><b>Ignore</b> the failed connection error message.</li>
+                            <li>And then <b>copy and paste</b> the full URL you are redirected to in here:</li>
+                        </ul>
+                        <input type="text" className={LOGIN2_INPUT_CLASS} onChange={this.handleURLInput} />
+                    </div>
+                }
+                {this.state.authStatus === AuthStatus.SUCCESS &&
+                    <div>
+                        <p>Authentication successful!</p>
+                    </div>
+                }
+                {this.state.authStatus === AuthStatus.FAILURE &&
+                    <div>
+                        <p>Authentication failed!</p>
+                    </div>
+                }
+            </div>
+        );
+    }
+}
+
+
+export async function showLogin2Dialog() {
```

Comment:
I think it's cleaner for components to have a default export. Maybe you can move this function to the file where it's called.

Reply:
I am sticking to the convention used in the other components with `showXXXDialog` function in the file.